tests/test_profiling.py

```diff
@@ -0,0 +1,77 @@
+"""Test the xdem.profiling functions."""
+
+from __future__ import annotations
+
+import glob
+import os.path as op
+
+import pandas as pd
+import pytest
+
+import xdem
+from xdem.profiler import Profiler
+
+
+class TestProfiling:
+
+    # Test that there's no crash when giving profiling configuration
+    @pytest.mark.parametrize(
+        "profiling_configuration", [(False, False), (True, False), (False, True), (True, True)]
+    )  # type: ignore
+    @pytest.mark.parametrize("profiling_function", ["load", "attribute", "coreg"])  # type: ignore
+    def test_profiling_configuration(self, profiling_configuration, profiling_function, tmp_path) -> None:
+        """
+        Test the all combinaisons of profiling with three examples of profiled functions.
+        """
+        s_gr = profiling_configuration[0]
+        s_rd = profiling_configuration[1]
+
+        Profiler.enable(save_graphs=s_gr, save_raw_data=s_rd)
+
+        dem = xdem.DEM(xdem.examples.get_path("longyearbyen_ref_dem"))
+        if profiling_function == "coreg":
+            xdem.coreg.VerticalShift().fit_and_apply(dem, dem)
+        if profiling_function == "attribute":
+            xdem.terrain.slope(dem=dem.data, resolution=dem.res[0])
+        Profiler.generate_summary(tmp_path)
+
+        if s_rd or s_gr:
```

Comment:
Nice tests!
My only remark is that we should add some line-to-line comments so that future us can understand it more quickly. Here it took me some time to dive into it, even though the tests are pretty simple.

For example, before `if s_rd or s_gr:`, we can add "If both "save_" arguments as passed, we check that both output files exists"
Then same later on for other `if` or `else` :slightly_smiling_face:
And also (and especially) in spots that took a while to figure out and could be important again later on. For instance the `Profiler.reset()` at the end seems important, but it's unclear why!